pyfr/plugins/integrate.py

```diff
@@ -130,7 +130,7 @@ def _eval_exprs(self, intg):
         pnames = self.elementscls.privarmap[self.ndims]
 
         # Iterate over each element type in the simulation
-        for i, (soln, eleinfo) in enumerate(zip(intg.soln, self.eleinfo)):
+        for i, (soln, grad_soln, eleinfo) in enumerate(zip(intg.soln, intg.grad_soln, self.eleinfo)):
```

Comment:
I'd only pull the property inside of the gradpnames block; avoids computing the gradient if it is not needed.

Reply:
Yes you're right